pkg/controller/plan/kubevirt.go

```diff
@@ -857,6 +859,48 @@ func (r *KubeVirt) GetGuestConversionPod(vm *plan.VMStatus) (pod *core.Pod, err
 	return
 }
 
+func (r *KubeVirt) GetVirtV2VDiskConfig(vm *plan.VMStatus, pod *core.Pod, step *plan.Step) (err error) {
+
+	if pod.Status.PodIP == "" {
+		//we need the IP for fetching the fimware info.
+		return
+	}
+
+	url := fmt.Sprintf("http://%s:8080/firmware", pod.Status.PodIP)
+	resp, err := http.Get(url)
+
+	if err != nil {
+		if strings.Contains(err.Error(), "connection refused") {
+			return nil
+		}
+		return
+	} else {
+		defer resp.Body.Close()
+		firmaware, errFirrmware := io.ReadAll(resp.Body)
```

Comment:
I think it makes more sense to return the entire OVF from the pod and let the caller takes what it needs, in this case the firmare

Reply:
I understand your perspective, but I'm not entirely convinced. It seems unnecessary to send the entire file when some of its content is irrelevant. In my opinion, it's more logical to parse it on the "server" side and transmit only the required fields. If we need more configuration in the future, we can always include it in the response or consider sending the full XML. However, based on our current requirements, sending everything seams, IMO redundant.

Reply:
i commented below as a general comment and then i saw this.
it's true that for now only `firmware` and having single value sent by the server is enough - because that's the change and we don't need the whole XML.
but, if you thinking about the future, in case we need more fields, i think sending the whole XML and parsing it within the controller is better. then, instead of changing the server to add more single fields, parsing, see how you send it to the controller - you won't need to.
you will need only an additional parse on controller side.

Reply:
ack, I'm getting your view on that, so i"ll change that to match your suggestion.

Reply:
we don't have to add an endpoint per attribute, we can make one that's generic and receives it as an arguments returning the value

Reply:
ideally, we should have taken the OVF that is produced by virt-v2v and adjust it according to network+storage mappings, vm name, etc. we could have also let virt-v2v to do that and even post the VM eventually (that was discussed with rjones long ago, but it's problematic in terms of permissions the pod needs to have and it breaks the abstraction we have in the migration flow..)
from that perspective, it's not like we should take "just a single or handful of properties" but the entire configuration. practically, we just need the firmware (and in oVirt [we also updated the display type](https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/exportimport/UpdateConvertedVmCommand.java#L113-L116) which is irrelevant for forklift). yet, I'd just return the entire OVF - the conversion pod already does quite a bit of work, no need to have the parsing of the OVF there as well...

Reply:
the code changed so now i'n sending the xml file and getting the relevant data from the controller.